### PR TITLE
Allow null Protostream fields

### LINKED ISSUE #30761: Protobuf deserialization has a default of an empty String
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Acceptance criteria

* Adding toggling the new behavior in protostream library
* Protostream release with the new feature
* Adapting the Keycloak code base to use the toggle
* Consider a soon-ish productized version of protostream (make it part of this issue, or create a new issue for KC26)

### Describe the bug

In protocol buffers v3 the default value of a String is an empty string. Due to that in the current implementation, all properties of a serialized objected with are null strings will become an empty string.

### Version

main

### Regression

- [X] The issue is a regression

### Expected behavior

A null String field should still be null after serialization and de-serialization. 

### Actual behavior

The value will become an empty String instead. 

In commits 66dd9e65b97b06321b3d07856e1bc56c267d1184 and 67098f04694cf0e06a4bd25c477b5110313a7228, this has already been fixed by treating any empty string as a null value. 

We are still missing it in other places like for example the brokerSessionId in the UserSessionEntity. 

### Anything else?

In a team discussion on Jun 25 I raised my concern that the test suite of Keycloak is not prepared to find those test failures as they will appear only when using the new external Infinispan setup where values are always serialized, or in a setup with multiple Keycloak nodes where keys are read from and written to another node. 

To keep the existing behavior of `MarshallUtil.marshallString` which kept null values, we were discussed with @pruivo, @ryanemerson, @mhajas and @kami619 to ask for a similar behavior in protostream. Ideally we would like to have a central switch to have all String values treated as null by default. 

Once the change is done, `Marshalling.emptyStringToNull` should no longer be necessary and can therefore be removed.

### PR DESCRIPTION
**Pull request by ryanemerson**
Closes #30761

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -100,6 +100,7 @@
         <hibernate-orm.plugin.version>6.2.13.Final</hibernate-orm.plugin.version>
         <hibernate.c3p0.version>6.2.13.Final</hibernate.c3p0.version>
         <infinispan.version>15.0.5.Final</infinispan.version>
+        <protostream.version>5.0.6.Final</protostream.version>
```
Comment (ahus1) [inline pom.xml:103]: Suggestion: Create a follow-up issue to align the version of protostream here with the one used in Infinispan once it catches up, and then remove the explicit dependency management we use below.

Comment (ryanemerson) [reply pom.xml:103]: I've created #31032 to track this

**Review by pruivo — APPROVED** [review]

**Review by ahus1 — DISMISSED** [review]
Thank you for this PR! Approving based on @pruivo's review.

**Comment by ahus1** [comment]
Added one small comment in the `pom.xml` to link to the tracking issue eaef2ec4c2c0641ef6697db50ca870901b6cc97d

**Comment by ryanemerson** [comment]
Thanks @ahus1

File: pom.xml [hunk 2]
```diff
@@ -100,6 +100,9 @@
         <hibernate-orm.plugin.version>6.2.13.Final</hibernate-orm.plugin.version>
         <hibernate.c3p0.version>6.2.13.Final</hibernate.c3p0.version>
         <infinispan.version>15.0.5.Final</infinispan.version>
+        <!-- Remove the explicit protostream dependency once Infinispan has caught up.
+        See https://github.com/keycloak/keycloak/issues/31032 that is tracking this -->
+        <protostream.version>5.0.7.Final</protostream.version>
```
Comment (pruivo) [inline pom.xml:105]: we cannot remove this since we need the version for the plugin.

Comment (pruivo) [reply pom.xml:105]: except if you want to re-introduce `infinispan.maven-plugins.version`.

Comment (ryanemerson) [reply pom.xml:105]: Argh good point @pruivo, I didn't connect the dots. I'll remove this comment and close the issue.

**Review by ahus1 — APPROVED** [review]
Thank you for double-checking and sorry for the confusion.

**Pull request merged**